server/src/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3147,12 +3147,11 @@ private boolean isPermissible(Long accountDomainId, Long offeringDomainId) {
                 ex.addProxyObject(template.getUuid(), "templateId");
                 throw ex;
             }
-            if (caller.getType() == Account.ACCOUNT_TYPE_DOMAIN_ADMIN) {
+            if (!template.isPublicTemplate() && !template.isFeatured() && caller.getType() == Account.ACCOUNT_TYPE_DOMAIN_ADMIN) {
```

Comment:
@kishankavala : I have made the changes as per your suggestion. Please check new code.